lldb/source/Plugins/Language/Swift/SwiftUnsafeTypes.cpp

```diff
@@ -263,6 +343,70 @@ bool SwiftUnsafeRawBufferPointer::Update() {
   return true;
 }
 
+class SwiftUnsafePointer final : public SwiftUnsafeType {
+public:
+  SwiftUnsafePointer(ValueObject &valobj, UnsafePointerKind kind);
+  bool Update() override;
+};
+
+SwiftUnsafePointer::SwiftUnsafePointer(ValueObject &valobj,
+                                       UnsafePointerKind kind)
+    : SwiftUnsafeType(valobj, kind) {}
+
+bool SwiftUnsafePointer::Update() {
+  if (!m_valobj.GetNumChildren())
+    return false;
+
+  // Here is the layout of Swift's Unsafe[Mutable]Pointer.
+  //
+  // ▿ Unsafe[Raw]Pointer
+  //   ▿ _rawValue : Int
+  //       - pointerValue : Int
+  //
+
+  CompilerType type = m_valobj.GetCompilerType();
+  if (!type.IsValid()) {
+    LLDB_LOG(GetLogIfAllCategoriesSet(LIBLLDB_LOG_DATAFORMATTERS),
+             "{0}: Couldn't get the compiler type for the "
+             "'Swift.UnsafePointer' ValueObject.",
+             __FUNCTION__, type.GetTypeName());
+    return false;
+  }
+
+  SwiftASTContext *type_system = GetSwiftASTContextTypeSystem(type);
```

Comment:
Instead of adding `GetSwiftASTContextTypeSystem()` here, it would be better to hoist `GetGenericArgumentType()` into the `TypeSystemSwift` base class and provide a default implementation using ReconstructType to `TypeSystemSwiftTypeRef`  (Or, even better, proivide a real implementation). Then you can just cast `type.GetTypeSystem(`) to `TypeSystemSwift` here and use that.